### PR TITLE
Build with hyprland-session.service

### PR DESCRIPTION
**Pull request by izmyname**
#### Describe your PR, what does it fix/add?

Hyprland is now built with hyprland-session.service

#### Is there anything you want to mention? 

Using both meson and cmake vars in hyprland-session.service looks hacky, but it's the best solution I could come up with, given my experience.

#### Is it ready for merging, or does it need work?
Ready

### CODE REVIEW

File: systemd/hyprland-session.service.in [hunk 1]
```diff
@@ -9,8 +9,8 @@ After=graphical-session-pre.target
 
 [Service]
 Type=notify
-ExecStart=/usr/bin/Hyprland
-ExecStop=/usr/bin/hyprctl dispatch exit
-ExecStopPost=/usr/bin/systemctl --user unset-environment DISPLAY WAYLAND_DISPLAY HYPRLAND_INSTANCE_SIGNATURE XDG_CURRENT_DESKTOP
+ExecStart=@CMAKE_INSTALL_PREFIX@@prefix@/bin/Hyprland
```
Comment (fufexan) [inline systemd/hyprland-session.service.in:12]: Could be easier to only leave `@bindir@/Hyprland` and use `set(bindir ${CMAKE_INSTALL_BINDIR})` in CMakeLists.txt.

File: systemd/hyprland-session.service.in [hunk 2]
```diff
@@ -9,8 +9,8 @@ After=graphical-session-pre.target
...
+ExecStop=@CMAKE_INSTALL_PREFIX@@prefix@/bin/hyprctl dispatch exit
+ExecStopPost=@CMAKE_INSTALL_PREFIX@@prefix@/bin/systemctl --user unset-environment DISPLAY WAYLAND_DISPLAY HYPRLAND_INSTANCE_SIGNATURE XDG_CURRENT_DESKTOP
```
Comment (fufexan) [inline systemd/hyprland-session.service.in:14]: This line is not correct, as systemd doesn't live in Hyprland's prefix. It should simply be `systemctl ...` and systemd should find it through PATH.

File: systemd/meson.build [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  install_dir: join_paths(get_option('datadir'), 'wayland-sessions'),  
+  install_tag: 'runtime',                                              
+)                                                                      
+
+prefix = get_option('prefix')
+conf_data = configuration_data()
+conf_data.set('prefix', prefix)
+
+
+  configure_file(
```
Comment (fufexan) [inline systemd/meson.build:12]: Weird indent here.

File: systemd/hyprland-session.service.in (hunk 1 above)
Comment (izmyname) [reply systemd/hyprland-session.service.in:12]: for some reason, both bindir from meson and CMAKE_INSTALL_BINDIR default to just /bin

Comment (fufexan) [reply systemd/hyprland-session.service.in:12]: ah yea, then use `@prefix@/@bindir@`

Comment (izmyname) [reply systemd/hyprland-session.service.in:12]: Okay, workin on it

Comment (izmyname) [reply systemd/hyprland-session.service.in]: done

File: CMakeLists.txt [hunk 4]
```diff
@@ -15,7 +15,10 @@ include(GNUInstallDirs)
 set(HYPRLAND_VERSION ${VER})
 set(PREFIX ${CMAKE_INSTALL_PREFIX})
 set(INCLUDEDIR ${CMAKE_INSTALL_INCLUDEDIR})
+set(bindir ${CMAKE_INSTALL_BINDIR})
+set(prefix ${CMAKE_INSTALL_PREFIX})
```
Comment (fufexan) [inline CMakeLists.txt:19]: Oh, we already seem to have PREFIX. Let's not duplicate things then, just make bindir -> BINDIR and remove prefix.

File: systemd/hyprland-session.service.in [hunk 5]
```diff
@@ -9,8 +9,8 @@ After=graphical-session-pre.target
 
 [Service]
 Type=notify
-ExecStart=/usr/bin/Hyprland
-ExecStop=/usr/bin/hyprctl dispatch exit
-ExecStopPost=/usr/bin/systemctl --user unset-environment DISPLAY WAYLAND_DISPLAY HYPRLAND_INSTANCE_SIGNATURE XDG_CURRENT_DESKTOP
+ExecStart=@prefix@/@bindir@/Hyprland
+ExecStop=@prefix@/@bindir@/hyprctl dispatch exit
```
Comment (fufexan) [inline systemd/hyprland-session.service.in:13]: These should be @UPPERCASE@ to match what we already have in CMake.

File: systemd/meson.build [hunk 6]
```diff
@@ -0,0 +1,17 @@
+install_data(                                                          
+  'hyprland-systemd.desktop',                                                  
+  install_dir: join_paths(get_option('datadir'), 'wayland-sessions'),  
+  install_tag: 'runtime',                                              
+)                                                                      
+
+prefix = get_option('prefix')
+bindir = get_option('bindir')
```
Comment (fufexan) [inline systemd/meson.build:8]: No need for these assignments.

File: systemd/meson.build [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+conf_data = configuration_data()
+conf_data.set('prefix', prefix)
+conf_data.set('bindir', bindir)
```
Comment (fufexan) [inline systemd/meson.build:11]:
```suggestion
conf_data.set('PREFIX', get_option('prefix'))
conf_data.set('BINDIR', get_option('bindir'))
```

File: CMakeLists.txt (hunk 4 above)
Comment (izmyname) [reply CMakeLists.txt:19]: A moment

**Review by fufexan — APPROVED** [review]
LGTM

**Pull request merged**